src/os/bluestore/BlueStore.cc

```diff
@@ -18299,7 +18323,34 @@ void BlueStore::_log_alerts(osd_alert_list_t& alerts)
   } else if (!spillover_alert.empty()){
     spillover_alert.clear();
   }
-
+  if (cct->_conf->bluestore_slow_op_event_counter_threshold) {
+    _trim_slow_op_event_queue(mono_clock::now());
+    if (!slow_op_event_queue.empty()) {
+      ostringstream ss;
+      uint64_t qsize = slow_op_event_queue.size();
+      ss << "got "
```

Comment:
May be better not to mention "more than N" in the alert message at all?
IMO "experiencing slow operations at BlueStore" is sufficient.

Reply:
Then we won't need the queue. Also N won't have any significance. We just need the last occurence of slow op/stalled read.

Reply:
queue is an internal mean to collect N events which result in a warning. I.e. this is a way to filter noise out. And this to be left as-is. But I'm not sure we should present this mechanics in the warning message...

Reply:
By the way do you only mean to remove "More than N" clause or the whole "N" thing in the message?

Reply:
In the warning message I think it's better to say something generic, e.g "experiencing slow operations at BlueStore". And that's all. Not sure it makes sense to describe all these conditions which triggered the warning: amount of events, time perios etc. This information is available in the log and just overburden the message. IMHO.

Reply:
Okay then we can just get rid the queue and simplify the message.

Reply:
> I think the only thing here we are using queue to get the number how many events has been occurred in last T period. Otherwise if we are just putting the warning without acknowledging the number, the we won't need the queue. The warning will persist for T period anyway as waning only depends on the last occurence. But I went for the queue approach just for the false positive indication. Let me know your thoughts.

From my experience we can get occasional "slow operation observed" in the log. And this can be ignored most of the time. So the users should have the ability to do that. 
Much worse condition is when periodic series of such indications appear. Users shouldn't miss that. But highly likely  they don't care much about how many indication they've got: 30, 300 or 3000.  If the alert flaps or is raised permanently - this indicates a pretty severe problem. 
Hence the rationale of having the queue (IMO) is to be able to filter out single indications but still being able to detect serial ones (and warn on them). It would be impossible without the queue.
Nevertheless the default setting is to warn on a single indication right now. Just to be as much careful as we can. Though I'm not 100% confident that's always the best approach from users perspective - hence we make this configurable.

Reply:
> Okay then we can just get rid the queue and simplify the message.
Simplify the message - yes,
get rid off the queue - nope

One is unable to detect event series  if needed without the queue..

Reply:
My question is in our current approach if it is serial indication, the output is specifically raising a warning for T period and that T period will always start from the last time the event occurred isn't it? Even if there is series of event the warning matters the last event only unless we want to share the information how many events occurred in the last T period.
Here is the example,
```
let's say T = 30, 
event1 --> timestamp 40 (warning raised)
event2 --> timestamp 50 (warning keep raised)
from timestamp (80+ to 85-) warning will be removed.
event3 --> timestamp 85 (warning raised again)
```
As we are not sharing information about how many event occurred how queue is helping? Isn't it always depends on the last occurrence?